File: navigation/src/main/java/com/example/navigation/stepsEngine/flow/rules/comparable/NullRule.java
package com.example.navigation.stepsEngine.flow.rules.comparable;

import com.example.navigation.stepsEngine.payment.FlowState;

public class NullRule extends ComparableRule {

    @Override
    public boolean evaluate(FlowState flowState) {
        return getField().get(flowState) == null;
    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/flow/rules/comparable/NullRule.kt
package com.example.navigation.stepsEngine.flow.rules.comparable

import com.example.navigation.stepsEngine.payment.FlowState

class NullRule() : ComparableRule() {

    constructor(mField: String) : this() {
       field = mField
    }
    override fun evaluate(flowState: FlowState): Boolean {
        return if (isNot)
            getValue(flowState) != null
        else
            getValue(flowState) == null
    }
}